### PR TITLE
BaSP-TG-63: add update admins redux

### PR DESCRIPTION
**Pull request by Licamus**
Add and update admins with redux

### CODE REVIEW

File: src/Components/Admins/AdminForm/AdminForm.jsx [hunk 1]
```diff
@@ -4,19 +4,25 @@ import Modal from '../../Shared/Modal';
...
-  const [adminId, setAdminId] = useState();
+  const [name, setName] = useState('');
+  const [lastName, setLastName] = useState('');
+  const [email, setEmail] = useState('');
+  const [password, setPassword] = useState('');
   const [edit, setEdit] = useState(false);
   const [showModal, setShowModal] = useState(false);
   const [isActionModal, setIsActionModal] = useState(false);
-  const [serverError, setServerError] = useState();
+  const admins = useSelector((state) => state.admins.list);
```
Comment (lmalvasia) [inline src/Components/Admins/AdminForm/AdminForm.jsx:22]: I think that we could have only one selector:
```suggestion
  const { list: admins, isLoading, error } = useSelector((state) => state.admins);
```

File: src/Components/Admins/AdminForm/AdminForm.jsx [hunk 2]
```diff
@@ -68,58 +74,51 @@ const AdminForm = () => {
...
-      setName(data.data.name);
-      setLastName(data.data.lastName);
-      setEmail(data.data.email);
-      setPassword(data.data.password);
-    } else {
-      setEdit(false);
+    try {
+      if (id) {
+        setEdit(true);
+        setName(admins.find((item) => item._id === id).name);
```
Comment (lmalvasia) [inline src/Components/Admins/AdminForm/AdminForm.jsx:80]: you are doing always the same find. We should save the admin in a new variable and reuse it.
```suggestion
        const currentAdmin = admins.find(item => item._id === id);
        setName(currentAdmin);
```

File: src/Components/Admins/AdminForm/AdminForm.jsx (hunk 1 above)
Comment (Licamus) [reply src/Components/Admins/AdminForm/AdminForm.jsx:22]: done!

File: src/redux/admins/thunks.js [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+      .then((response) => response.json())
+      .then((response) => {
+        if (response.error) {
+          throw new Error(response.message);
+        } else {
+          return dispatch(postAdminSuccess(response.data));
+        }
+      })
+      .catch((err) => {
+        dispatch(postAdminError(err.toString()));
```
[the remark this answers is not in the document]
Comment (Licamus) [reply src/redux/admins/thunks.js:99]: done!

File: src/Components/Admins/AdminForm/AdminForm.jsx (hunk 2 above)
Comment (Licamus) [reply src/Components/Admins/AdminForm/AdminForm.jsx]: done!

**Review by lmalvasia — APPROVED** [review]

File: src/Components/Admins/AdminForm/AdminForm.jsx [hunk 4]
```diff
@@ -68,58 +72,52 @@ const AdminForm = () => {
...
+        setLastName(currentAdmin.lastName);
+        setEmail(currentAdmin.email);
+        setPassword(currentAdmin.password);
+      } else {
+        setEdit(false);
+      }
+    } catch (error) {
+      console.log(error);
     }
   }, []);
```
Comment (baudraccoverona) [inline src/Components/Admins/AdminForm/AdminForm.jsx:89]:
```suggestion
  }, [id]);
```

**Pull request merged**